simulatingrisk/hawkdovevar/model.py

```diff
@@ -13,18 +17,94 @@ def set_risk_level(self):
         # generate a random risk level
         self.risk_level = self.random.randint(0, num_neighbors)
 
+    def play(self):
+        super().play()
+        # when enabled by the model, periodically adjust risk level
+        if self.model.adjustment_round:
+            self.adjust_risk()
+
+    @property
+    def most_successful_neighbor(self):
+        """identify and return the neighbor with the most points"""
+        # sort neighbors by points, highest points first
+        # adapted from risky bet wealthiest neighbor
+        return sorted(self.neighbors, key=lambda x: x.points, reverse=True)[0]
+
+    def adjust_risk(self):
+        # look at neighbors
+        # if anyone has more points
+        # either adopt their risk attitude or average theirs with yours
+
+        best = self.most_successful_neighbor
+        # if most successful neighbor has more points and a different
+        # risk attitude, adjust
+        if best.points > self.points and best.risk_level != self.risk_level:
+            # adjust risk based on model configuration
+            if self.model.risk_adjustment == "adopt":
+                # adopt neighbor's risk level
+                self.risk_level = best.risk_level
+            elif self.model.risk_adjustment == "average":
+                # average theirs with mine, then round to a whole number
+                # since this model uses discrete risk levels
+                self.risk_level = round(
+                    statistics.mean([self.risk_level, best.risk_level])
+                )
```

Comment:
Very, very cool, and slick. I remember talking about this and/or similar strategies of risk adjustment with David and Lara and you.